src/beast/beast/nudb/detail/bulkio.h

```diff
@@ -45,10 +45,16 @@ class bulk_reader
     bulk_reader (File& f, std::size_t offset,
         std::size_t last, std::size_t buffer_size);
 
+    std::size_t
+    offset() const
+    {
+        return offset_ - avail_;
```

Comment:
I don't understand this. Current position minus bytes left to read doesn't sound like something interesting.